### ISSUE TITLE
ToC plugin add unnecessary empty line 

### ISSUE THREAD

**Issue by exoego** [tinymce/tinymce#5369]
**What is the current behavior? Describe the bug**

It seems that ToC plugin add `<li>&nbsp;</li>` between an item for `<h3>` and `<h1>`.
These `<li>&nbsp;</li>` are shown as empty line with bullet, causing visual glitch.
![image](https://user-images.githubusercontent.com/127635/73323649-76d85d80-428b-11ea-9521-6229107df532.png)

**Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tinymce.com](http://fiddle.tinymce.com/) or similar.**

Create a document something like this:

```
ADD TOC HERE

1.    // this should be H1
1.2.  // this should be H2
1.2.3 // this should be H3
2.     // this should be H1
```

Then show the created document (not in tinymce editor, but in as plain HTML).

**What is the expected behavior?**

`<li>&nbsp;</li>`  should not be inserted and no empty line shown in ToC.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**

* 5.1.5
* Firefox and

**Comment by lostkeys** [comment]
Yeah that's not intended. I can [reproduce it in this fiddle](http://fiddle.tinymce.com/0ahaab/1). Thx for reporting!

**Comment by rianazara** [comment]
I see the issue is still open. Is this defect being looked at? When can we expect a fix for this in Version 4?
